### ISSUE TITLE
Use scene.pick to get GroundPrimitive and return wrong results

### ISSUE THREAD

**Issue by abcdweiok** [CesiumGS/cesium#9753]
[Sandcastle](https://sandcastle.cesium.com/index.html#c=pVZtU+o4FP4rXWZ2bK8YKAKKgLtcRAVBFNGriOOUNkCgTbpJiwt3/O97+kpBmH3zC03yPOc5bzkxk6ljQVwLNRlFBh5rrunUdB0L0WdzTKWqdICXrenoSidd0mo+rprqLWmKJu0V9Hqz2Jzbz0/1VgkB6A/jau6B5re5x4/O6tLq1lXrJdc4bvfv1cHFvfMym+RvYW9wMTU7q4E1mM3zt/05addb9gCMdWaPBe/75fmedGeNXGemFzqrDmBAoLcoNvXV+/S73rgY9dRitnY8u2vdj6fG4OYHe++0ijdscXty3bM/DspDutC4tCD4A3MIgeIPKQzzyd+ThyndX9cZdTRCMR+mFGAFPKFjioEW8JG/DE3qmoW5Bmf+JgqW4RmhwtEo2IXj17eIgKnj+/Cqqqcony+WSqVs8aSYP8mepKVcHuVPi/lCoZRTT0v5Qs6jjRmXfXtVtSyRSjWXhZ/DQ0X6OaRSaPA1+wY219+HkiyTXwtStVqVVEX6TTrKoqwqnUnwk80VlLKU4KoJrrqDG3A8spcToDEqmImRySbyQRzQQTq0EYA8j20miEMA7cXrbSbdPfLtqukN5WDrH0CP9kAP/xf0XzgQQj3kmx9wXG5ku2IqJ3rsCjMLO3zZDBHyz8A+Mc6kYSriHQ1TYJSE2pOQc5Zs1jtmLieMRvYiO97fAnMH/3nJuKU5ZzEe80izzkzGa7aNNe4t0VOj1288v192e51aP722YwcS1wT6mOvTnfrxoRzu1zUQF0Sjx2jMmXWBJxxjUeNcW8pxDyhKpPKpxIKa43AycoF9JiWi0T1v4zB837eTWIuYvqQPkZN49NJot7s/lERsYso+NuJ5gI29dmWHu3jts/fxqXi9DZ/BbZ9w5lLjjhMLIlxA4TXD2Kj75nlQrsmWHgQed47vqxYXaTP3e2sZtYFuMoGhpTy/A2/BV9/jYCyhsbnsswBtYOEQqnmFWad5ZxXl7VtQ+Hq3YCufzWZ9vXDOCWxi3cFGOlHieKBONWqY25P4QQc5+mBrOm4sQOA6AMnRZAWm8KyHZCSw06S2C6+TF4U8dqn/IckWW2ALDIQD0h9ERJ9jozuagU/xsPY2YzCK+nTHiPOAwYBL2gmAZBx1HTyW8HAY8gZGibo6842MpY0zRAx/xEaZUtYXgGPH5bQcLD+/ZcJxAQa2tGJugrzON/KvEXi+414gh31fAkZWIpm9IltOJ7UifdDYQpW/urMNsqOLgSbY2XsNxTrI2Kbn5C/J1xgR6g3AuqkJQcZE9xtbSU6Uv8uK1/E96CxmycqX3MTZCaaAhE2Bd1R+rbHO0X+qBsh8pvfdi/7SxqjduOy/19vN+o1STqVTFeEsTXweefs7sWzGHcnlpoxQxsGWbWpgPjNyIf8O0oWIQqtkktSKQRbwLFV3/DsE4wWSCydj1zQfyAoPU+eVDOC/UE2mGYROulARU1t6sKl63g42EUKVDCx3Mx3GzJHGtyz/BQ)

![2](https://user-images.githubusercontent.com/13530397/130589464-01b4586e-1a2a-41fc-9210-c0046fcabd00.gif)

**Comment by GatorScott** [comment]
Quick look showed that it works properly if only one row is generated.  Adding even one more (i <= 6) results in failure.

**Comment by abcdweiok** [comment]
> Quick look showed that it works properly if only one row is generated. Adding even one more (i <= 6) results in failure.

Yes, Now I get all pickedObjects through `scene.drillPick`, and then take the last one. I don't know if it's the right thing to do, or if there's a better way.

**Comment by ZenDay** [comment]
For more help: 
With OrthographicFrustum, we can drillpick successfully.(only when camre.pitch equals to -90 degree).

[sandcastle](https://sandcastle.cesium.com/index.html#c=pVZtU+o4FP4rXWZ2bK8YKAKKgLteRAVBFNGriOOUNkCgbbpJCgt3/O97+kpBmH3zi03yPOc5bzkhk6lhTlwLNaiNDDzSXFNc6DrmvEdn2Jaq0gFeNifDa510SLPxtGqod6TBG3a3oNcaxcbMeXmuNUsIQH8Y1zMPNLvLPS3aqyurU1Ot11z9uNV7UPuXD+J1Os7fwV7/cmK2V32rP53l73oz0qo1nT4Ya0+fCt7368sD6UzrufZUL7RXbcCAQHdebOirj8l3vX457KrF7MXx9L75MJoY/dsf9KPdLN7S+d3JTddZHJQH9lxj0pzgBWYQgo0XUhjms78nD1K6v65RW2jExmyQUoAV8LiObQy0gI/8ZWhS1yzMNDjzN1GwDM+IzYVmg104fnv3rWUyksuJPZY6TEzomGnOhOhXzOXCtQZ2wEZ8QYQ+6dEdGHntFOgJP5g3VT1F+XyxVCpliyfF/En2JC3l8ih/WswXCqWcelrKF3Ke/ogy2XesqpYlUqnmsvDv8FCRfg5sKTT4ln0Hm+vvQ0mWya8FqVqtSqoi/SYdZVFWlc4k+JfNFZSylOCqCa66gxtwPLIXB9CozamJkUnH8kEc0EE6tBGAPI8dyokggPbi9TaT7h75dtX0hnKw9Q+gR3ugh/8L+i8cCKEe8t0POO4b5Lh8Iiea9RpTCwu2bIQI+Wdgnxhn0iAV8Y4GKTBKQu1xyDlLdv09NZdjakf2Ijve3xwzgf+8oszSxFmMxyzSrFGTsgvHwRrzlui53u3VXz6uOt32RS+9tuMEEjcEWprpk5368aEc7tc0EOdEs4/RiFHrEo8ZxvyCMW0pxz2gKJHKpxILakIwMnSBfSYlotE9b+MwfN+3k3gRMX1JHyIn8ei13mp1fiiJ2PiELjbieYSNvXZlwVy89tn7+FS83obPYGyMGXVt454RCyKcQ+E1w9io++Z5UK7xlh4EHneO76sWF2kz93trGbWBblKOoaU8vwNvwVff43BCjcxljwZoA3NBbM0rzDrNO6sob9+Cwte7BVv5bDbr64VTlGMT6wIb6USJ4yE40WzD3B7pjzrI2Y+OpuP6HARuApAcjWhgcs96SEYci4btuPDMeVHII9f2PyTZonNsgYFwQPqDiOgzbHSGU/ApnvreZgxGUZ/uGHEeMBhwSTsBkIyiroNXF14gQ97AKFFXZ76RkbRxhojhj9goU8r6AjAsXGaXg+Xnt0w4LsDAllbMTZDX+Ub+NQLPd9wLJOj3JWBkJZLZK7LldFIr0geNLVT5qzvbICe6GGiMxd5ryNdBxjY9J39JPuuI2N4ArJka52REdL+xleRE+buseB3fhc6i8Fh/yU2cnWAKSNjkeEfl1xrrHP2naoDMZ3rfvegtHYxa9aveR63VqN0q5VQ6VeFiaeLzyNvfieVQJiSXmTJCGYEtx9TAfGboQv4F0jmPQqtkktSKQebwLFV3/K6C8QLJhZORa5qPZIUHqfNKBvBfqCbVDPix1IGKmNrSg03U81awiRCqZGC5mykoNYca27L8Fw)

**Comment by ggetz** [comment]
Also reported in https://github.com/CesiumGS/cesium/issues/10863.

**Comment by MeanYou** [comment]
Hi, do you have a plan to fix the problem? @ggetz

**Comment by ggetz** [comment]
@MeanYou There hasn't been any activity on this item recently. 

If you'd like to add more information about your use case, that may help us prioritize.

If you are interesting in contributing, please let us know and we'd be happy to discuss implementation or review a PR. 

Thanks!

**Comment by 1716247670** [comment]
I also encountered the same problem. When I created the GroundPrimitive, I found that its bounding box is a cube, but the Primitive is a sphere.
